src/v/datalake/translation/state_machine.cc

```diff
@@ -59,8 +62,14 @@ ss::future<> translation_stm::do_apply(const model::record_batch& batch) {
           record.release_value());
         vlog(
           _log.trace,
-          "updating highest translated offset to {}",
-          value.highest_translated_offset);
+          "updating highest translated offset to {}[]",
+          value.highest_translated_offset,
+          value.highest_translated_timestamp);
+        if (
+          value.highest_translated_timestamp > _highest_translated_timestamp) {
+            update_highest_translated_timestamp(
+              value.highest_translated_timestamp);
+        }
```

Comment:
maybe we can make it a simple `_highest_translated_timestamp = std::max(value.highest_translated_timestamp, _highest_translated_timestamp)` ?

Reply:
done